pytensor/graph/basic.py

```diff
@@ -1088,20 +1088,20 @@ def truncated_graph_inputs(
         return truncated_inputs
 
     blockers: Set[Variable] = set(ancestors_to_include)
+    # variables that go here are under check already, do not repeat the loop for them
+    seen: Set[Variable] = set()
     # enforce O(1) check for node in ancestors to include
     ancestors_to_include = blockers.copy()
 
     while candidates:
         # on any new candidate
         node = candidates.pop()
-
-        # There was a repeated reference to this node, we have already investigated it
-        if node in truncated_inputs:
+        # we've looked into this node already
+        if node in seen:
             continue
-
         # check if the node is independent, never go above blockers;
         # blockers are independent nodes and ancestors to include
-        if node in ancestors_to_include:
+        elif node in ancestors_to_include:
```

Comment:
Nitpick: no need to change

```suggestion
        if node in ancestors_to_include:
```

Reply:
I decided to go with elif so there is a single if else clause